packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+  const PreparedMouseCursor();
+}
+
+/// A mouse cursor that does nothing when activated.
+///
+/// The singleton instance of this class is available at
+/// [SystemMouseCursors.uncontrolled], which also introduces its usage.
+/// Directly instantiating this class is unnecessary and thus not allowed, since
+/// it's not configurable.
+class NoopMouseCursor extends PreparedMouseCursor {
+  // Application code shouldn't directly instantiate this class, since its only
+  // instance is accessible at [SystemMouseCursors.releaseControl].
+  const NoopMouseCursor._();
+
+  @override
+  String get debugDescription => '';
+}
+
+/// A mouse cursor that comes with the system that the program is running on.
+///
+/// System cursors are the most commonly used cursors, since they are avaiable
+/// without external resources, and matches the experience of native apps.
+/// Examples of system cursors are a pointing arrow, a pointing hand, a double
+/// arrow for resizing, or a text I-beam, etc.
+/// 
+/// An instance of [SystemMouseCursor] refers to one cursor from each platform
+/// that represents the same concept, such as text, being clickable, or forbidden
+/// operation. Since the set of system cursors supported by each platform varies,
+/// multiple instances can correspond to the same system cursor.
+///
+/// [SystemMouseCursors] enumerates the complete set of system cursors supported
+/// by Flutter, which are hard-coded in the platform engine. Therefore, manually
+/// instantiating this class is neither useful nor supported.
+class SystemMouseCursor extends PreparedMouseCursor {
+  // Application code shouldn't directly instantiate system mouse cursors, since
+  // the supported system cursors are enumerated in [SystemMouseCursors].
+  const SystemMouseCursor._({
+    @required this.shapeCode,
+    @required this.debugDescription,
+  }) : assert(shapeCode != null),
+       assert(debugDescription != null);
+
+  /// A globally unique number that identifies the shape of the cursor.
+  ///
+  /// The exact value of a [shapeCode] is intentionally random, and its
+  /// interpretation (i.e. corresponding system cursor) is hard-coded in the
+  /// platform engine.
+  ///
+  /// See the documentation of [SystemMouseCursor] for introduction.
+  final int shapeCode;
+
+  @override
+  final String debugDescription;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is SystemMouseCursor
+        && other.shapeCode == shapeCode;
+  }
+
+  @override
+  int get hashCode => shapeCode;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(IntProperty('shapeCode', shapeCode, level: DiagnosticLevel.debug));
+  }
+}
+
+/// A collection of system [MouseCursor]s.
+///
+/// System cursors are mouse cursors provided by the current platform, available
```

Comment:
```suggestion
/// System cursors are standard mouse cursors that are provided by the current platform. They don't require external resources.
```
They're probably the most commonly used cursors, but how common they are doesn't matter here.